### PR TITLE
[aot] Support matrix/vector for compute graph in Python

### PR DESCRIPTION
**Pull request by Zilize**

### CODE REVIEW

File: taichi/program/matrix.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                  const DataType dtype,
+                  const intptr_t &data)
+      : length_(length), dtype_(dtype), data_(data) {
+  }
+
+  DataType dtype() const {
+    return dtype_;
+  }
+
+  uint32_t length() const {
```
Comment (ailzhang) [inline taichi/program/matrix.h:22]: nit: better indicate whether the length is in bytes or # of elements

File: taichi/python/export_lang.cpp [hunk 2]
```diff
@@ -622,55 +624,96 @@ void export_lang(py::module &m) {
...
+#undef PER_C_TYPE
+              default:
+                TI_ERROR("Unsupported scalar type {}", type_id);
+            }
+          } else {
+            TI_NOT_IMPLEMENTED;
+          }
+        }
+        self->jit_run(compile_config, args);
+        matrix_buffers.clear();
```
Comment (ailzhang) [inline taichi/python/export_lang.cpp:714]: nit: do you need to clear these manually?

File: tests/python/test_graph.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 supported_floating_types = [ti.f32] if platform.system() == "Darwin" else [ti.f32, ti.f64]
 
-supported_archs_cgraph = [ti.vulkan, ti.opengl]
+supported_archs_cgraph = [ti.vulkan]
```
Comment (ailzhang) [inline tests/python/test_graph.py:12]: hmmm did you remove ti.opengl for unsupported data types?

**Review by ailzhang — APPROVED** [review]
Nice!

File: taichi/program/matrix.h (hunk 1 above)
Comment (Zilize) [reply taichi/program/matrix.h:22]: Solved with a comment.

File: taichi/python/export_lang.cpp (hunk 2 above)
Comment (Zilize) [reply taichi/python/export_lang.cpp]: Solved.

File: tests/python/test_graph.py (hunk 3 above)
Comment (Zilize) [reply tests/python/test_graph.py]: Done.

**Pull request merged**